src/Table/write_fits.cxx

```diff
@@ -234,32 +213,53 @@ void tablator::Table::write_fits(
                                        combined_labeled_trailing_info_lists.begin(),
                                        combined_labeled_trailing_info_lists.end());
 
+    combined_labeled_properties.insert(combined_labeled_properties.end(),
+                                       combined_labeled_attributes.begin(),
+                                       combined_labeled_attributes.end());
+
+
+    // labeled_properties are stored as keywords whose values are strings
+    // of the form
+    // "value XMLATTR_DOTatt1:val1 XMLATTR_DOTatt2:val2 ... XMLATTR_DOTattn:valn".
 
     for (const auto &label_and_prop : combined_labeled_properties) {
-        auto keyword_mapping = fits_keyword_mapping(true);
-        std::string name = label_and_prop.first;
+        auto keyword_ucd_mapping = fits_keyword_ucd_mapping(true);
+        std::string label = label_and_prop.first;
         const auto &prop = label_and_prop.second;
-        auto i = keyword_mapping.find(name);
-        if (i != keyword_mapping.end()) {
-            name = i->second;
-        }
-        std::string comment, value(prop.get_value());
+
+        std::string value(prop.get_value());
+
+        // JTODO sort of duplicates flatten_property().
+        std::string comment;
+
         for (auto &att : prop.get_attributes()) {
+#ifdef FIXED_FITS_COMMENT
+            // This step prepares us to store the comment in a special FITS way,
+            // but as of 13Nov20, comments will be truncated or omitted if
+            // comment.size() + value.size() > 65.
+
             if (att.first == "comment") {
-                comment = att.second;
-            } else if (att.first != "ucd") {
+                comment.assign(att.second);
+            } else
+#endif
+                    if (att.first == "name" && !att.second.empty() &&
+                        !boost::ends_with(label, XMLATTR)) {
+                // hack to make 'label' values distinct when we really want
+                // them to be XXX.INFO
+                label += DOT + XMLATTR_DOT + att.second;
+            }
+            if (att.first != "ucd") {
+                // read_fits() sets 'ucd' value based on 'name' value. (!)
```

Comment:
I still don't understand why 😄. Also, doesn't this approach end up throwing out UCDs that don't have an associated keyword in the result of `fits_keyword_ucd_mapping(true)`?

Reply:
Yes, it does.  This treatment of UCDs is more or less inherited from the master branch, but I gather that weeding out undocumented UCDs is a bug and not a feature.  Will fix.